drivers/tty/serdev/pibridge.c

```diff
@@ -437,7 +437,10 @@ int pibridge_req_io(u8 addr, u8 cmd, u8 *snd_buf, u16 snd_len, u8 *rcv_buf,
 				"receive crc error in io-req\n");
 			return -EIO;
 		}
-		crc = pibridge_crc8(crc, rcv_buf, pkthdr.len);
+
+		if (to_receive)
+			crc = pibridge_crc8(crc, rcv_buf, pkthdr.len);
```

Comment:
Why not move this into the existing `if (to_receive)` branch further up in the function?

I.e., receive `to_receive` bytes, calculate the checksum over it, then receive `to_discard` bytes. Granted, you'll waste some cycles for checksum calculation in the `to_discard` case, but that's an error condition anyway which shouldn't happen, so we can probably afford wasting a few extra cycles there.